smart_control/dataset/dataset_partition_test.py

```diff
@@ -0,0 +1,468 @@
+  def test_partition_data(self):
+    data = self.partition.data
+    self.assertIsInstance(data, np.lib.npyio.NpzFile)
+
+    # fmt: off
+    # we are surfacing each key into its own high-level public property:
+    with self.subTest('action_value_matrix'):
+      np.testing.assert_array_equal(
+          data['action_value_matrix'],
+          self.partition.action_value_matrix
+      )
+
+    with self.subTest('observation_value_matrix'):
+      np.testing.assert_array_equal(
+          data['observation_value_matrix'],
+          self.partition.observation_value_matrix,
+      )
+
+    with self.subTest('reward_value_matrix'):
+      np.testing.assert_array_equal(
+          data['reward_value_matrix'],
+          self.partition.reward_value_matrix
+      )
+
+    with self.subTest('reward_info_value_matrix'):
+      np.testing.assert_array_equal(
+          data['reward_info_value_matrix'],
+          self.partition.reward_info_value_matrix,
+      )
+    # fmt: on
+
+  @unittest.skipUnless(TEST_DATASET, SKIP_REASON)
+  def test_partition_metadata(self):
+    metadata = self.partition.metadata
+
+    self.assertIsInstance(metadata, dict)
+    expected_keys = [
+        'action_ids_map',
+        'action_timestamps',
+        'observation_ids_map',
+        'observation_timestamps',
+        'reward_ids_map',
+        'reward_info_timestamps',
+        'reward_timestamps',
+    ]
+    self.assertEqual(sorted(metadata.keys()), expected_keys)
+
+    # fmt: off
+    # pylint: disable=line-too-long
+    # we are surfacing each key into its own high-level public property:
+    self.assertEqual(metadata['action_ids_map'], self.partition.action_ids_map)
+    self.assertEqual(metadata['observation_ids_map'], self.partition.observation_ids_map)
+    self.assertEqual(metadata['reward_ids_map'], self.partition.reward_ids_map)
+
+    self.assertEqual(metadata['action_timestamps'], self.partition.action_timestamps)
+    self.assertEqual(metadata['observation_timestamps'], self.partition.observation_timestamps)
+    self.assertEqual(metadata['reward_timestamps'], self.partition.reward_timestamps)
+    self.assertEqual(metadata['reward_info_timestamps'], self.partition.reward_info_timestamps)
+    # pylint: enable=line-too-long
+    # fmt: on
+
+  #
+  # DATA PROPERTIES...
+  #
+
+  @unittest.skipUnless(TEST_DATASET, SKIP_REASON)
+  def test_action_value_matrix(self):
+    self.assertEqual(self.partition.action_value_matrix.shape, (51852, 3))
+
+  @unittest.skipUnless(TEST_DATASET, SKIP_REASON)
+  def test_observation_value_matrix(self):
+    self.assertEqual(self.partition.observation_value_matrix.shape, (51852, 1198))  # pylint: disable=line-too-long
+
+  @unittest.skipUnless(TEST_DATASET, SKIP_REASON)
+  def test_reward_value_matrix(self):
+    # QUESTION: there are 3252 reward identifiers, but only 17 columns here
+    # so what are these 17 columns about?
+    self.assertEqual(self.partition.reward_value_matrix.shape, (51852, 17))
```

Comment:
QUESTION / ISSUE: there are 3252 reward identifiers, and 3252 columns in the `reward_info_value_matrix`, but only 17 columns in the `reward_value_matrix`. So what are these 17 columns about? Or is there an issue with the dataset?

@JudahGoldfeder